src/dataflow/components/dataflow-program.tsx

```diff
@@ -42,9 +42,25 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   private programEditor: NodeEditor;
   private programEngine: any;
 
+  private components = [new NumberReteNodeFactory(numSocket),
+                        new MathReteNodeFactory(numSocket),
+                        new TransformReteNodeFactory(numSocket),
+                        new LogicReteNodeFactory(numSocket),
+                        new SensorReteNodeFactory(numSocket),
+                        new RelayReteNodeFactory(numSocket),
+                        new GeneratorReteNodeFactory(numSocket)];
```

Comment:
There's no reason to store these persistently with the object. They can be accessed later via `this.programEditor.components` as you are doing in `addNode()` below. This array can stay local to `initProgramEditor()`.

Reply:
Fixed.